### PR TITLE
avoid creating archive twice and determine owner after archive.open

### PR DESCRIPTION
**Pull request by clkao**

### CODE REVIEW

**Comment by juliangruber** [comment]
+1, the double archive thing confused me before

File: app.js [hunk 1]
```diff
@@ -12,56 +12,67 @@ var explorer = require('./')
 var $hyperdrive = document.querySelector('#hyperdrive-ui')
 var $shareLink = document.getElementById('share-link')
 
-var url = window.location.toString()
-var key = url.split('#')[1]
-updateShareLink()
-var archive = drive.createArchive(key, {live: true})
-swarm(archive)
+var [, key, file] = window.location.hash.substr(1).match('([^\/]+)(/?.*)')
```
Comment (juliangruber) [inline app.js]: destructuring assignment isn't cross browser ready yet unfortunately

Comment (clkao) [inline app.js]: Sad. Is there something lightweight that does transpiling alongside browserify?

Comment (juliangruber) [inline app.js]: you can pick the parts of babel that you need but in my experience it's not worth it yet, just slows you down too much

Comment (mafintosh) [inline app.js]: :-1: on any transpiling

**Comment by okdistribute** [comment]
nice

**Pull request merged**